### PR TITLE
Fixed bugs, improved accuracy.

### PR DESCRIPTION
**Pull request by ironwallaby**
This PR does the following:

*   All code is now operating under Strict Mode. *[style]*
*   Refactor Julian date conversion into a separate library so it can easily be tested. *[style]*
*   Switched from `assert` to `chai`, since it's much more commonly used with Mocha. *[style]*
*   Told Git to stop whining that `node_modules/` wasn't in the repository. *[style]*
*   Fix global variable leak in function `meanphase`. *[bug]*
*   Remove all references to the `(year, month, day, hour, minute, second)` mode of the Date constructor. This version of the constructor is not consistent across platforms and should never be used. *[bug]*
*   Work around `process.env.TZ` bug in Node, preventing the tests from always working. *[bug]*
*   *Greatly* improved the accuracy of the library (from ~0.6% average moon phase error to ~0.02% average moon phase error) by fixing erroneous Julian date conversions. *[bug]*
*   Added tests for moon phase accuracy and correctness of Julian date conversions. *[enhancement]*
*   Made existing tests respect (at least to a first approximation) reasonable bounds on accuracy. *[enhancement]*
*   Optimized conversion from Julian date to JavaScript Date object. *[performance]*

I also bumped the version number to 0.2.1: these are all improvements and bug fixes, and do not affect the library's interface in any way, shape, or form.

### CODE REVIEW

**Comment by ironwallaby** [comment]
I should note that this PR fixed #9, #10, #11.

**Comment by ironwallaby** [comment]
Interesting, looks like you're using `const` in the base module. That's a little fishy unless you restrict Node to recent versions: I'll switch to standard constants for the time being and update this PR.

**Comment by ironwallaby** [comment]
OK, all fixed.

**Comment by ryanseys** [comment]
:heart_eyes:

File: lib/julian.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  function from_date(date) {
+    return date.getTime() / 86400000 + 2440587.5;
+  }
+
+  function to_date(julian) {
+    return new Date((julian - 2440587.5) * 86400000);
+  }
+
+  exports.from_date = from_date;
```
Comment (ryanseys) [inline lib/julian.js]: Can you use camelCasing which is more standard?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lune",
-  "version": "0.2.0",
+  "version": "0.2.1",
```
Comment (ryanseys) [inline package.json]: Can you remove this for now. I will bump the version in a separate commit when I publish to npm. It makes it easier to track when versions are bumped.

File: test/index.js [hunk 3]
```diff
@@ -1,40 +1,113 @@
...
+            ["1989-09-22T02:10Z", 0.75],
+            ["1989-09-29T21:47Z", 0.00]
+          ];
 
-      assert.equal(JSON.stringify(feb17), JSON.stringify(lunedata));
+      var error = 0;
+      var i;
+      var obs;
+      var e;
+      for(i = 0; i < observations.length; i++) {
```
Comment (ryanseys) [inline test/index.js]: for (var i = 0;

File: test/index.js [hunk 4]
```diff
@@ -1,40 +1,113 @@
...
+        obs = observations[i];
+        e = Math.abs(lune.phase(new Date(obs[0])).phase - obs[1]);
+        if(e > 0.5) {
```
Comment (ryanseys) [inline test/index.js]: nit: `if (e > 0.5)`

Comment (ryanseys) [inline test/index.js]: Forget this comment, I'll perhaps consider using `standard` style later.

File: test/index.js (hunk 3 above)
Comment (ironwallaby) [inline test/index.js]: I can make that change, but be advised it's very bad style to do that: it implies that `var i` is scoped to that block, when, in fact, it isn't!

If we bump the version of Node that Travis is testing, I'd be happy to do a pass through to switch to using `let`, in which case the code will be a lot cleaner :)

File: package.json (hunk 2 above)
Comment (ironwallaby) [inline package.json]: Done, good sir.

File: lib/julian.js (hunk 1 above)
Comment (ironwallaby) [inline lib/julian.js]: Done, good sir. I was confused, since camel case was not used consistently in the original code!

File: test/index.js (hunk 3 above)
Comment (ryanseys) [inline test/index.js]: Oooh, yeah let's do that. (I haven't been keeping up with all the changes Node has been making). You can bump the Node version in the travis yaml file. Ideally we test against the latest stable Node version as well as the LTS version.

**Comment by ryanseys** [comment]
@ironwallaby want me to merge this now, publish, and then we can make the new node changes in a separate change / version?

**Comment by ironwallaby** [comment]
Yes, please. I think that bumping the Node version should be 0.3.0 at least, since it involves breaking changes. (Specifically, people can't use it on Node 0.10.0 any more without specifying runtime flags!)

**Comment by ryanseys** [comment]
Thanks, merged and published as 0.2.1.

**Pull request merged**